src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/RMIHelper.java

```diff
@@ -78,34 +78,36 @@ public static void rebind(String uniqueID, Remote object) throws DebuggerExcepti
         }
     }
 
-    public static void unbind(String uniqueID) throws DebuggerException {
-        String name = getName(uniqueID);
+    public static void unbind(String serverID, String serverName) throws DebuggerException {
+        String name = getName(serverID, serverName);
         try {
             Naming.unbind(name);
         } catch (Exception exp) {
             throw new DebuggerException(exp);
         }
     }
 
-    public static Remote lookup(String debugServerID) throws DebuggerException {
-        // debugServerID follows the pattern [unique_id@]host[:port]
-        // we have to transform this as //host[:port]/<serverNamePrefix>['_'<unique_id>]
+    public static Remote lookup(String connectionString) throws DebuggerException {
+        // connectionString follows the pattern [serverid@]host[:port][/servername]
+        // we have to transform this as //host[:port]/<servername>['_'<serverid>]
+        Matcher matcher = CONNECT_PATTERN.matcher(connectionString);
+        matcher.find();
 
-        int index = debugServerID.indexOf('@');
-        StringBuilder nameBuf = new StringBuilder("//");
-        String uniqueID = null;
-        if (index != -1) {
-            nameBuf.append(debugServerID.substring(index + 1));
-            uniqueID = debugServerID.substring(0, index);
-        } else {
-            nameBuf.append(debugServerID);
+        String rmiObjectName = matcher.group("servername") == null ? DEFAULT_RMI_OBJECT_NAME
+                                                                   : matcher.group("servername");
+        String serverNamePrefix = System.getProperty("sun.jvm.hotspot.rmi.serverNamePrefix");
+        if (serverNamePrefix != null) {
+            System.err.println("WARNING: sun.jvm.hotspot.rmi.serverNamePrefix is deprecated. Please specify it in --connect.");
+            rmiObjectName = serverNamePrefix;
```

Comment:
If the servername was already specified with `--connect`, I would think that we would not want to override it with the property if it is also set. Maybe if both are set then fail with an error. Same in `getName()` below.